unicodetools/src/test/java/org/unicode/text/UCD/TestCodeInvariants.java

```diff
@@ -230,4 +242,132 @@ private static String showInfo(
+        Set<Object> nonCollidingProperties = new HashSet<>();
+        nonCollidingProperties.addAll(Arrays.asList(UcdProperty.values()));
+        nonCollidingProperties.addAll(
+                Arrays.asList(UcdPropertyValues.General_Category_Values.values()));
+        nonCollidingProperties.addAll(Arrays.asList(UcdPropertyValues.Script_Values.values()));
+        nonCollidingProperties.remove(UcdProperty.ISO_Comment); // Collides with gc Other.
+        nonCollidingProperties.remove(UcdProperty.Case_Folding); // Collides with gc Format.
+        nonCollidingProperties.remove(
+                UcdProperty.Lowercase_Mapping); // Collides with gc Cased_Letter.
+        nonCollidingProperties.remove(UcdProperty.Script); // Collides with gc Currency_Symbol.
+        testLM3NamespaceUniqueness(
+                nonCollidingProperties,
+                x ->
+                        x instanceof UcdProperty
+                                ? ((UcdProperty) x).getNames().getAllNames()
+                                : ((Named) x).getNames().getAllNames(),
+                Set.of("Age"),
+                "Unusual (not a violation of UnicodeSet requirement): New gc-sc-non-binary property namespace collision");
+    }
+
+    <T> void testLM3NamespaceUniqueness(
+            Iterable<T> namespace,
+            Function<T, List<String>> getNames,
+            Set<String> expectedRedundant,
+            String message) {
+        final Map<String, T> entitiesByAlias = new TreeMap<>(UnicodeProperty.PROPERTY_COMPARATOR);
+        final Map<String, String> aliasesByLM3Skeleton = new HashMap<>();
+        for (T entity : namespace) {
+            for (String alias : getNames.apply(entity)) {
+                final var matchingEntity = entitiesByAlias.get(alias);
+                final var lm3Skeleton = UnicodeProperty.toSkeleton(alias);
+                final var matchingAlias = aliasesByLM3Skeleton.get(lm3Skeleton);
+                assertTrue(
+                        matchingEntity == null || entity.equals(matchingEntity),
+                        message
+                                + ": alias "
+                                + alias
+                                + " for "
+                                + entity
+                                + " matches alias "
+                                + matchingAlias
+                                + " for "
+                                + entitiesByAlias.get(alias));
+                if (matchingEntity != null && !expectedRedundant.contains(alias)) {
+                    assertEquals(
+                            matchingAlias,
+                            alias,
+                            "Unusual (not a stability policy violation): distinct aliases for "
+                                    + entity
+                                    + " match each other");
+                }
+                entitiesByAlias.put(alias, entity);
```

Comment:
optional:

putIfAbsent()?
https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Map.html#putIfAbsent(K,V)

or don't put() if  matchingEntity != null  ?

Reply:
For this one, if it is present, we have just checked that it is consistent with what is already there, so it doesn’t matter. So this comes down to expressing intent/readability.
On the one hand `put` feels like the default operation, so might as well use that if it does not matter.
But maybe `putIfAbsent` makes it clearer that we never mutate the map?

For the other one, `put` could be mutating (if we have an expectedRedundant alias), and we probably should not mutate (the earlier alias should be the canonical one).

So let’s say `putIfAbsent` for both.